cns/service/main.go

```diff
@@ -544,7 +544,11 @@ func main() {
 			DebugMode:                    ts.DebugMode,
 		}
 
-		logger.InitAI(aiConfig, ts.DisableTrace, ts.DisableMetric, ts.DisableEvent)
+		if aiKey := cnsconfig.TelemetrySettings.AppInsightsInstrumentationKey; aiKey != "" {
+			logger.InitAIWithIKey(aiConfig, aiKey, ts.DisableTrace, ts.DisableMetric, ts.DisableEvent)
+		} else {
```

Comment:
Just a NIT: might be better to set the default in setTelemetrySettingDefaults and not have to do this if/else scenario

Reply:
I tried your suggestion, but the problem I found was that the default value is defined as an unexported variable in cns/logger, so there was no good way for me to set the default in the `setTelemetrySettingDefaults` function becuase there is no way for me to read that value without at least exporting the variable. And the problem with _that_, is that we inject the value of that string at build-time using `go build`, and I'm not confident that I can enumerate all of the places where that value gets set, since we build CNS in a few different places in a few different pipelines. That's all to say - I think we'll do a big telemetry refactor soon once we have a path forward without AI, so I'm kind of ok with merging this as-is (especially if you feel the feedback is a nit).